engine/src/main/java/com/odysseusinc/arachne/executionengine/execution/r/DockerOverseer.java

```diff
@@ -58,17 +59,29 @@ public DockerOverseer(
         }));
         pos = stdout.length();
         this.client = client;
-        init.thenAccept(containerId ->
-                executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS)
-        );
+        init.thenAccept(containerId -> {
+            this.containerId = containerId;
+            executor.scheduleWithFixedDelay(this::writeLogs, updateInterval, updateInterval, TimeUnit.MILLISECONDS);
+        });
         this.init = init;
     }
 
     @Override
     public CompletableFuture<ExecutionOutcome> abort() {
-        return result.isDone() ? result : CompletableFuture.completedFuture(
-                new ExecutionOutcome(Stage.ABORT, "Abort is not yet supported for docker analysis", null)
-        );
+        return CompletableFuture.supplyAsync(() -> {
+            if (result.isDone()) {
+                return result.join();
+            } else {
+                try {
+                    client.stopContainerCmd(containerId).exec();
+                    client.removeContainerCmd(containerId).withRemoveVolumes(true).exec();
+                    return new ExecutionOutcome(Stage.ABORT, "Docker container aborted successfully", null);
```

Comment:
Defect: empty stdout in 3rd argument will overwrite everything in datanode. ExecutionOutcome must return FULL stdout.
Defect: this is not a correct termination state. First argument is stage and second argument is error that happend on that staget preventing the process from progressing. So, stage="ABORT" / error="something" signals that aborting has failed. A correct  outcome here is stage="ABORTED" / error=null. The message about successful abort should be placed in the stdout.